### PR TITLE
Add support for client feature flags to be passed as query parameters

### PR DESCRIPTION
**Pull request by groszewn**
Some plugins (e.g. [image](https://github.com/tensorflow/tensorboard/blob/963afb7be883c15373c121c96c9039828e3fb728/tensorboard/plugins/image/tf_image_dashboard/tf-image-loader.ts#L356) and [audio](https://github.com/tensorflow/tensorboard/blob/963afb7be883c15373c121c96c9039828e3fb728/tensorboard/plugins/audio/tf_audio_dashboard/tf-audio-loader.ts#L111)) specify URLs as src attributes on some HTML elements, which precludes them from attaching the necessary client feature flags via the `requestManager`. This PR adds support to the client feature flags middleware to also accept feature flags via query parameters, so that we can support this use case.

### CODE REVIEW

File: tensorboard/backend/client_feature_flags.py [hunk 1]
```diff
@@ -39,25 +44,66 @@ def __init__(self, application):
...
+        potential_feature_flags = query_string_json.get(
+            "tensorBoardFeatureFlags", []
         )
-        context.set_in_environ(environ, ctx)
+        if not potential_feature_flags:
+            return {}
+        try:
+            client_feature_flags = json.loads(potential_feature_flags[0])
+        except json.JSONDecodeError:
+            return {}
```
Comment (bmd3k) [inline tensorboard/backend/client_feature_flags.py:104]: Similar to the header-parsing logic, should we consider making this an InvalidArgumentError? Client provided tensorBoardFeatureFlags but it was malformed. Probably due to programmer error. They'd probably want to know.

File: tensorboard/backend/client_feature_flags.py [hunk 2]
```diff
@@ -39,25 +44,66 @@ def __init__(self, application):
...
 
-        return self._application(environ, start_response)
+        if not isinstance(client_feature_flags, dict):
+            return {}
```
Comment (bmd3k) [inline tensorboard/backend/client_feature_flags.py:107]: Should we also consider making this an InvalidArgumentError? Also similar to the header-parsing logic.

**Review by bmd3k — APPROVED** [review]

File: tensorboard/backend/client_feature_flags.py (hunk 1 above)
Comment (groszewn) [reply tensorboard/backend/client_feature_flags.py]: Yep, that definitely makes sense at this level. Updated to raise an error similar to the header path.

File: tensorboard/backend/client_feature_flags.py (hunk 2 above)
Comment (groszewn) [reply tensorboard/backend/client_feature_flags.py]: Yep, here as well. Updated.

**Pull request merged**